Jellyfin.Server.Implementations/StorageHelpers/StorageHelper.cs

```diff
@@ -0,0 +1,100 @@
+using System.IO;
+using MediaBrowser.Common.Configuration;
+using MediaBrowser.Model.System;
+using Microsoft.Extensions.Logging;
+
+namespace Jellyfin.Server.Implementations.StorageHelpers;
+
+/// <summary>
+/// Contains methods to help with checking for storage and returning storage data for jellyfin folders.
+/// </summary>
+public static class StorageHelper
+{
+    private const long TwoGigabyte = 2147483647L;
+    private const long FiveHundredAndTwelveMegaByte = 536_870_911L;
+    private static string[] _byteHumanizedSuffixes = ["B", "KB", "MB", "GB", "TB", "PB", "EB"];
+
+    /// <summary>
+    /// Tests the available storage capacity on the jellyfin paths with estimated minimum values.
+    /// </summary>
+    /// <param name="applicationPaths">The application paths.</param>
+    /// <param name="logger">Logger.</param>
+    public static void TestCommonJfPathsForStorageCapacity(IApplicationPaths applicationPaths, ILogger logger)
+    {
+        TestDataDirectorySize(applicationPaths.DataPath, logger, TwoGigabyte);
+        TestDataDirectorySize(applicationPaths.LogDirectoryPath, logger, FiveHundredAndTwelveMegaByte);
+        TestDataDirectorySize(applicationPaths.CachePath, logger, TwoGigabyte);
+        TestDataDirectorySize(applicationPaths.ProgramDataPath, logger, TwoGigabyte);
+        TestDataDirectorySize(applicationPaths.TempDirectory, logger, TwoGigabyte);
+    }
+
+    /// <summary>
+    /// Gets the free space of a specific directory.
+    /// </summary>
+    /// <param name="path">Path to a folder.</param>
+    /// <returns>The number of bytes available space.</returns>
+    public static FolderStorageInfo GetFreeSpaceOf(string path)
+    {
+        try
+        {
+            var driveInfo = new DriveInfo(path);
+            return new FolderStorageInfo()
+            {
+                Path = path,
+                FreeSpace = driveInfo.AvailableFreeSpace,
+                UsedSpace = driveInfo.TotalSize - driveInfo.AvailableFreeSpace,
+                StorageType = driveInfo.DriveType.ToString(),
+            };
+        }
+        catch
+        {
+            return new FolderStorageInfo()
+            {
+                Path = path,
+                FreeSpace = -1,
+                UsedSpace = -1,
+                StorageType = null,
+            };
+        }
+    }
+
+    private static void TestDataDirectorySize(string path, ILogger logger, long threshold = -1)
+    {
+        logger.LogDebug("Check path {TestPath} for storage capacity", path);
+        var drive = new DriveInfo(path);
+        if (threshold != -1 && drive.AvailableFreeSpace < threshold)
+        {
+            throw new InvalidOperationException($"The path `{path}` exceeds the minimum required free capacity of {HumanizeStorageSize(threshold)}");
+        }
+
+        logger.LogInformation(
+            "Storage path `{TestPath}` ({StorageType}) successfully tested with {FreeSpace} left free which is over the minimum of {MinFree}.",
```

Comment:
Similar as previous comment. Consider changing tested to checked. 

Also, "{FreeSpace} left free" doesnt read well. Consider something along the lines of: "Storage path `{TestPath}` ({StorageType}) was successfully checked. Free space: {FreeSpace}, which is above the required minimum of {MinFree}."?